packages/penrose-core/src/compiler/Style.ts

```diff
@@ -350,8 +354,10 @@ const checkRelPattern = (varEnv: Env, rel: RelationPattern): StyErrors => {
 
     // TODO(error)
     if (isErr(res1)) {
-      // TODO(error): extract and return error -- how to convert from SubstanceError to StyError?
-      return ["substance typecheck error in B"];
+      const subErr1: SubstanceError = unsafelyGetErr(res1);
```

Comment:
Use `res1.error` instead. The `if` check will make sure this is the right error type.